src/LinearRewardsErc4626.sol

```diff
@@ -207,13 +232,52 @@ abstract contract LinearRewardsErc4626 is ERC4626 {
         totalBalance += amount;
     }
 
+    function _computeFeeRaw(uint256 _assets) private view returns (uint256) {
+        return _assets * mintingFee / FEE_PRECISION;
+    }
+
+    function _computeFeeTotal(uint256 _assets) private view returns (uint256) {
+        return (_assets * mintingFee) / (mintingFee + FEE_PRECISION);
+    }
+
+    function _takeFee(uint256 _feeAmount) private {
+        if (_feeAmount > 0) {
+            asset.safeTransferFrom(msg.sender, FEE_RECIPIENT, _feeAmount);
+            emit FeeTaken(FEE_RECIPIENT, _feeAmount);
+        }
+    }
+
+    function previewDeposit(uint256 _assets) public view override returns (uint256) {
+        uint256 feeAmount = _computeFeeTotal(_assets);
+        return super.previewDeposit(_assets - feeAmount);
+    }
+    
     /// @notice The ```deposit``` function allows a user to mint shares by depositing underlying
     /// @param _assets The amount of underlying to deposit
     /// @param _receiver The address to send the shares to
     /// @return _shares The amount of shares minted
     function deposit(uint256 _assets, address _receiver) public override returns (uint256 _shares) {
         syncRewardsAndDistribution();
-        _shares = super.deposit({ assets: _assets, receiver: _receiver });
+
+        uint256 feeAmount = _computeFeeTotal(_assets);
+
+        uint256 assetsNoFee = _assets - feeAmount;
+
+        // Check for rounding error since we round down in previewDeposit.
+        /// @dev calling super.previewDeposit to ignore feeBPS
+        require((_shares = super.previewDeposit(assetsNoFee)) != 0, "ZERO_SHARES");
+
+        // Need to transfer before minting or ERC777s could reenter.
+        asset.safeTransferFrom(msg.sender, address(this), assetsNoFee);
+
+        _mint(_receiver, _shares);
+
+        // Emit _assets transferred from sender including fees
+        emit Deposit(msg.sender, _receiver, _assets, _shares);
+
+        afterDeposit(assetsNoFee, _shares);
+
+        _takeFee(feeAmount);
     }
```

Comment:
# QA - NIT - Change order of operations to move transfer to bottom

- We use eBTC
- eBTC is known and not ERC777
- Move transfer and fee to be at the end together

e.g.

```solidity
       
        _mint(_receiver, _shares);

        // Emit _assets transferred from sender including fees
        emit Deposit(msg.sender, _receiver, _assets, _shares);

        afterDeposit(assetsNoFee, _shares);

        // Ignore ERC777, we use eBTC
        asset.safeTransferFrom(msg.sender, address(this), assetsNoFee);

        _takeFee(feeAmount);
```